forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -747,24 +748,23 @@ public final Card moveTo(final Zone zoneTo, Card c, SpellAbility cause, Map<Abil
         // use FThreads.invokeInNewThread to run code in a pooled thread
         return moveTo(zoneTo, c, null, cause, params);
     }
-    public final Card moveTo(final Zone zoneTo, Card c, Integer position, SpellAbility cause) {
-        return moveTo(zoneTo, c, position, cause, AbilityKey.newMap());
-    }
 
     public final Card moveTo(final ZoneType name, final Card c, SpellAbility cause, Map<AbilityKey, Object> params) {
         return moveTo(name, c, 0, cause, params);
     }
-    public final Card moveTo(final ZoneType name, final Card c, final int libPosition, SpellAbility cause) {
-        return moveTo(name, c, libPosition, cause, AbilityKey.newMap());
-    }
+
     public final Card moveTo(final ZoneType name, final Card c, final int libPosition, SpellAbility cause, Map<AbilityKey, Object> params) {
         // Call specific functions to set PlayerZone, then move onto moveTo
         switch(name) {
             case Hand:          return moveToHand(c, cause, params);
             case Library:       return moveToLibrary(c, libPosition, cause, params);
             case Battlefield:   return moveToPlay(c, c.getController(), cause, params);
             case Graveyard:     return moveToGraveyard(c, cause, params);
-            case Exile:         return exile(c, cause, params);
+            case Exile:
+                if (!c.canExiledBy(cause, true)) {
+                    return null;
+                }
+                return exile(c, cause, params);
```

Comment:
looks like only MeldEffect needs tweaking because it still calls `exile(CardCollection)` directly

Reply:
i think i need to ask Judges what exactly happens if a meld pair can't be exiled

if it still tries to meld them or not

Reply:
yes but you would still exile them anyway
so now that could be prevented too...

Reply:
i added some fix, but i don't know if that handles all corner cases